library/sources/GraphQL.ts

```diff
@@ -0,0 +1,51 @@
+/* eslint-disable prefer-rest-params */
+import { getContext } from "../agent/Context";
+import { Hooks } from "../agent/hooks/Hooks";
+import { Wrapper } from "../agent/Wrapper";
+import type { ExecutionArgs } from "graphql/execution/execute";
+import { extractInputsFromDocument } from "./graphql/extractInputsFromDocument";
+
+export class GraphQL implements Wrapper {
+  private inspectGraphQLExecute(args: unknown[]): void {
+    if (!Array.isArray(args) || typeof args[0] !== "object") {
+      return;
+    }
+    const executeArgs = args[0] as ExecutionArgs;
+    const context = getContext();
+    if (!context) {
+      // We expect the context to be set by the wrapped http server
+      return;
+    }
+
+    const userInputs = extractInputsFromDocument(executeArgs.document);
+
+    if (
+      executeArgs.variableValues &&
+      typeof executeArgs.variableValues === "object"
+    ) {
+      for (const value of Object.values(executeArgs.variableValues)) {
+        if (typeof value === "string") {
+          userInputs.push(value);
+        }
+      }
+    }
+
+    if (userInputs.length > 0) {
+      if (Array.isArray(context.graphql)) {
+        context.graphql.push(...userInputs);
+      } else {
+        context.graphql = userInputs;
```

Comment:
feels a bit weird to mutate the context here? of course, we normally wrap functions in a different way to call `runWithContext`

I think it's fine for now